doc/man3/OSSL_PARAM_int.pod

```diff
@@ -231,9 +231,12 @@ will be assigned the size the parameter's I<data> buffer should have.
 
 OSSL_PARAM_get_utf8_string() retrieves a UTF8 string from the parameter
 pointed to by I<p>.
-The string is either stored into I<*val> with a size limit of I<max_len> or,
-in the case when I<*val> is NULL, memory is allocated for the string and
-I<max_len> is ignored.
+The string is stored into I<*val> with a size limit of I<max_len>, not
+counting a terminating NUL byte.  If the retrieved UTF8 string was
+shorter than I<max_len> bytes, a terminating NUL byte is added into
+I<*val>.
+If I<*val> is NULL, memory is allocated for the string and I<max_len>
+is ignored.
```

Comment:
This means the function has the infamous strncpy() semantics of not adding the NUL byte at the end if it does not fit. I am concerned about caller errors because of that. IMO we should fail in that case.

Reply:
So, better to go with strlcpy semantics?
This has me thinking, why don't we simply use OPENSSL_strlcpy() instead of reinventing the wheel yet again?

Reply:
The strlcpy semantics is IMO a little bit better in that it does not leave you with unterminated strings. But perhaps the most correct thing here would be to error out?

Reply:
Okie, I made that change

Reply:
... and that change actually broke things :-/

Reply:
Yeah, because the param tests do something that is invalid with the current semantics - setting and getting the same raw parameter sets with the same sizes on set and get. Either the tests have to be changed to not do this or the check could be made more lenient so it does not try to append the `\0` if there already is `\0` in the data and thus it does not have to have the space for the additional NUL byte?

Reply:
Not just with the current semantics, they are outright abusive, have always been and need to be reworked.

The lenient solution you're talking about might work.  We do have `OPENSSL_strnlen()`, which fits the purpose.  I'll try that for now.
The tests still need to become less abusive, but that's another rabbit hole that I would rather not have to deal with nownownow.

Reply:
Done

Reply:
Unfortunately the sanitizer build still detects something illegal.

Reply:
Ok, will look